mesonbuild/dependencies/base.py

```diff
@@ -2058,6 +2093,38 @@ def get_path(self):
     def get_name(self):
         return self.name
 
+class Msys2PathConverter():
+
+    def __init__(self):
+        self.cygpath = ExternalProgram('cygpath', silent=True)
```

Comment:
If we don't find cygpath then meson will stop completely, is that an okay behavior in this case?

Reply:
its ok， cygpath are the base component of msys

Reply:
> If we don't find cygpath then meson will stop completely, is that an okay behavior in this case?

silent=True means won't stop completely

Reply:
no, that's what `required` does. `silent` means don't print a found message on the command line.

Reply:
But anyway, I think that answers the question.

Reply:
The constructor of ExternalProgram doesn't have the required option, do you talking about the 
ExternalDependency class constructor?
```
class ExternalProgram:
    windows_exts = ('exe', 'msc', 'com', 'bat', 'cmd')
    # An 'ExternalProgram' always runs on the build machine
    for_machine = MachineChoice.BUILD

    def __init__(self, name: str, command: T.Optional[T.List[str]] = None,
                 silent: bool = False, search_dir: T.Optional[str] = None,
                 extra_search_dirs: T.Optional[T.List[str]] = None):
```